web/src/navbar_alerts.ts

```diff
@@ -112,11 +111,23 @@ export function dismiss_upgrade_nag(ls: LocalStorage): void {
     }
 }
 
-export function check_profile_incomplete(): boolean {
-    if (!current_user.is_admin) {
-        return false;
+export function should_show_organization_profile_incomplete_banner(timestamp: number): boolean {
+    const today = new Date(Date.now());
+    const time = new Date(timestamp * 1000);
+    const days_old = time_zone_util.difference_in_calendar_days(
+        today,
+        time,
+        timerender.display_time_zone,
+    );
+
+    if (days_old >= 15) {
+        return true;
     }
-    if (!should_display_profile_incomplete_alert(realm.realm_date_created)) {
+    return false;
+}
+
+export function is_organization_profile_incomplete(): boolean {
+    if (!current_user.is_admin) {
         return false;
     }
```

Comment:
I think the `is_admin` check belongs in `should_show_organization_profile_incomplete_banner`, just to make the behavior match the function names.

Reply:
Makes sense. Moved the `is_admin` check to `should_show_organization_profile_incomplete_banner` and updated the test cases.